packages/entity/src/internal/EntityDataManager.ts

```diff
@@ -236,4 +312,63 @@ export default class EntityDataManager<
     // TODO(wschurman): check for races with load
     await Promise.all(pairs.map(([key, value]) => this.invalidateOneAsync(key, value)));
   }
+
+  /**
+   * Invalidate all in-memory caches for sets of key-value pairs for all transactions and parent transactions.
+   * @param pairs - key-value pairs to invalidate
+   */
+  public invalidateKeyValuePairsForTransaction(
+    queryContext: EntityTransactionalQueryContext,
+    pairs: readonly LoadPair<TFields, TIDField, any, any, any>[],
+  ): void {
+    // TODO(wschurman): check for races with load
+
+    // invalidate all query contexts in transaction tree
+    const outermostTransactionalQueryContext =
+      EntityDataManager.getOutermostTransactionalQueryContextIfInNestedTransaction(queryContext);
+    const allQueryContextsToInvalidate = [
+      outermostTransactionalQueryContext,
+      ...EntityDataManager.getAllChildTransactionalQueryContexts(
+        outermostTransactionalQueryContext,
+      ),
+    ];
```

Comment:
Just curious: Does the order matter? I think it doesn't, especially because this isn't an async function so it's not going to get interleaved with another concurrent transaction.

Reply:
Correct. That's how I interpret it. Probably most theoretically correct would be `self -> parent -> outermost -> descendants` but since this is running synchronously it should be fine in any order.